apis/apps/v1alpha1/nebulacluster_common.go

```diff
@@ -285,11 +285,13 @@ func generateAgentContainer(c NebulaClusterComponent, init bool) corev1.Containe
 	}
 
 	agentImage := defaultAgentImage
-	if nc.Spec.Agent.Image != "" {
-		agentImage = nc.Spec.Agent.Image
-	}
-	if nc.Spec.Agent.Version != "" {
-		agentImage = fmt.Sprintf("%s:%s", agentImage, nc.Spec.Agent.Version)
+	if nc.Spec.Agent != nil {
+		if nc.Spec.Agent.Image != "" {
+			agentImage = nc.Spec.Agent.Image
+		}
+		if nc.Spec.Agent.Version != "" {
+			agentImage = fmt.Sprintf("%s:%s", agentImage, nc.Spec.Agent.Version)
+		}
```

Comment:
If only config the version, so the `agentImage` is `vesoft/nebula-agent:latest:v1.0.0` ?
```
spec:
    agent:
        version: v1.0.0
```